server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -173,23 +177,27 @@ private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
 
         // WARNING: This line is EXTREMELY IMPORTANT.
         // You MUST REMOVE candidates that are still in use
-        if (candidateMap.remove(relativePath) != null) {
-          log.debug("Candidate was still in use: {}", relativePath);
+        GcCandidate gcTemp = candidateMap.remove(relativePath);
+        if (gcTemp != null) {
+          log.debug("File Candidate was still in use: {}", relativePath);
+          inUseCandidates.add(gcTemp);
         }
 
         String dir = relativePath.substring(0, relativePath.lastIndexOf('/'));
-        if (candidateMap.remove(dir) != null) {
-          log.debug("Candidate was still in use: {}", relativePath);
+        GcCandidate gcT = candidateMap.remove(dir);
+        if (gcT != null) {
+          log.debug("Directory Candidate was still in use by file ref: {}", relativePath);
```

Comment:
```suggestion
          log.debug("Directory Candidate was still in use by file ref: {}", relativePath);
          // intentionally not adding directory candidates to inUseCandidates because dir candidates are only added once if deleted because in use nothing will add them again.
```